tools/test.py

```diff
@@ -138,6 +138,14 @@ def RunSingle(self, parallel, thread_id):
       try:
         start = datetime.now()
         output = case.Run()
+        # SmartOS has a bug that causes unexpected ECONNREFUSED errors.
+        # See https://smartos.org/bugview/OS-2767
+        # If ECONNREFUSED on SmartOS, retry the test one time.
+        if (output.UnexpectedOutput() and
+          sys.platform == 'sunos5' and 
+          'ECONNREFUSED' in output.output.stderr):
+            sys.stderr.write('ECONNREFUSED encountered on SmartOS; retrying test.')
```

Comment:
It actually should use some of the classes in this file to produce correct output, be it TAP or anything else. Right now it doesn't fit into the global scheme of `test.py`, this is why I am a bit worried about leaving it as it is.